scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerBackupRunner.java

```diff
@@ -109,32 +116,27 @@ private boolean someTaskIsRunning() {
 
     private void performBackup() throws IOException {
         String backupFileName = PREFIX + DateTime.now().toString("yyyy-MM-dd'T'HH-mm") + ".zip";
-        File destinationFolder = new File(destination);
-
-        if (!destinationFolder.isAbsolute()) {
-            destinationFolder = new File(PASchedulerProperties.SCHEDULER_HOME.getValueAsString(), destination);
-        }
 
-        if (!destinationFolder.exists()) {
-            destinationFolder.mkdirs();
+        if (!resolvedDestination.exists()) {
+            resolvedDestination.mkdirs();
         }
 
-        if (destinationFolder.exists() && destinationFolder.isDirectory() && destinationFolder.canWrite()) {
-            File backupFile = new File(destinationFolder, backupFileName);
+        if (resolvedDestination.exists() && resolvedDestination.isDirectory() && resolvedDestination.canWrite()) {
+            File backupFile = new File(resolvedDestination, backupFileName);
             LOGGER.info("Performing backup to " + backupFile);
             String[] foldersToZip = targets.stream()
                                            .map(target -> (new File(PASchedulerProperties.SCHEDULER_HOME.getValueAsString(),
                                                                     target)).getAbsolutePath())
                                            .toArray(String[]::new);
             ZipUtils.zip(foldersToZip, backupFile, null);
         } else {
-            LOGGER.error("Cannot save a backup because backup destination folder does not exist: " + destination);
+            LOGGER.error("Cannot save a backup because backup configuredDestination folder does not exist: " +
+                         configuredDestination);
         }
     }
 
     private void removeOldBackups() {
-        File fodlerWhereBackups = new File(destination);
-        File[] files = fodlerWhereBackups.listFiles();
+        File[] files = resolvedDestination.listFiles();
```

Comment:
I think this can throw an IOException on first run.

1) removeOldBackups is called before performBackup
2) performBackup calls mkdirs
so if the backup directory does not exist, on first run, an IOException will be thrown when reaching resolvedDestination.listFiles();

Reply:
ok, listFiles returns null if destination does not exist